File: 0406-queue-reconstruction-by-height/0406-queue-reconstruction-by-height.cpp
class Solution {
public:
    static bool cmp(vector<int> a, vector<int> b) {
        if (a[0] == b[0]) {
            return a[1] < b[1];
        }
        return a[0] > b[0];
    }
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        sort(people.begin(), people.end(), cmp);

        vector<vector<int>> res;

        for (vector<int> person : people) {
            res.insert(res.begin() + person[1], person);
        }
        
        return res;
    }
};

File: 0406-queue-reconstruction-by-height/0406-queue-reconstruction-by-height.py
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        people.sort(key=lambda x: (-x[0], x[1]))
        res = []
        
        for val, idx in people:
            res.insert(idx, [val, idx])
        
        return res